code/game/objects/gamemode/crash/items.dm

```diff
@@ -0,0 +1,219 @@
+		if(!lighthack)
+			flick("nuclearbombc", src)
+			icon_state = "nuclearbomb1"
+		extended = TRUE
+	return
+
+
+
+/obj/machinery/nuclearbomb/crash/Topic(href, href_list)
+	. = ..()
+	if(.)
+		return
+	if (!usr.canmove || usr.stat || usr.restrained())
+		return
+	if (!(usr.contents.Find(src) || (in_range(src, usr) && istype(loc, /turf))))
+		usr << browse(null, "window=nuclearbomb")
+		return
+
+	usr.set_interaction(src)
+	
+	if(href_list["r_auth"])
+		if (r_auth)
+			r_auth.loc = loc
+			has_auth = FALSE
+			r_auth = null
+		else
+			var/obj/item/I = usr.get_active_held_item()
+			if (istype(I, /obj/item/disk/nuclear/crash/red))
+				if(usr.drop_held_item())
+					I.forceMove(src)
+					r_auth = I
+	if(href_list["g_auth"])
+		if (g_auth)
+			g_auth.loc = loc
+			has_auth = FALSE
+			g_auth = null
+		else
+			var/obj/item/I = usr.get_active_held_item()
+			if (istype(I, /obj/item/disk/nuclear/crash/green))
+				if(usr.drop_held_item())
+					I.forceMove(src)
+					g_auth = I
+	if(href_list["b_auth"])
+		if (b_auth)
+			b_auth.loc = loc
+			has_auth = FALSE
+			b_auth = null
+		else
+			var/obj/item/I = usr.get_active_held_item()
+			if (istype(I, /obj/item/disk/nuclear/crash/blue))
+				if(usr.drop_held_item())
+					I.forceMove(src)
+					b_auth = I
+
+	if (has_auth)
+		if (href_list["time"])
+			var/time = text2num(href_list["time"])
+			timeleft += time
+			timeleft = min(max(round(timeleft), 60), 600)
+		if (href_list["timer"])
+			if (timing == -1.0)
+				return
+			if (safety)
+				to_chat(usr, "<span class='warning'>The safety is still on.</span>")
+				return
+			timing = !timing
+			if(timing)
+				GLOB.nukes_set_list |= src
+				start_processing()
+			if(!lighthack)
+				icon_state = (timing) ? "nuclearbomb2" : "nuclearbomb1"
+		if (href_list["safety"])
+			safety = !safety
+			if(safety)
+				timing = FALSE
+				GLOB.nukes_set_list -= src
+				stop_processing()
+			else
+				GLOB.nukes_set_list |= src
+				start_processing()
+		if (href_list["anchor"])
+			if(removal_stage == 5)
+				anchored = FALSE
+				visible_message("<span class='warning'> \The [src] makes a highly unpleasant crunching noise. It looks like the anchoring bolts have been cut.</span>")
+				return
+
+			anchored = !anchored
+			if(anchored)
+				visible_message("<span class='warning'> With a steely snap, bolts slide out of [src] and anchor it to the flooring.</span>")
+			else
+				visible_message("<span class='warning'> The anchoring bolts slide back into the depths of [src].</span>")
+
+	for(var/mob/M in viewers(1, src))
+		if ((M.client && M.interactee == src))
+			attack_hand(M)
+
+
+/obj/item/disk/nuclear/crash
+	name = "generic crash nuclear authentication disk"
+	desc = "Better keep this safe."
+	icon_state = "datadisk0"
+	w_class = WEIGHT_CLASS_TINY
+	resistance_flags = UNACIDABLE | INDESTRUCTIBLE
+
+/obj/item/disk/nuclear/crash/Initialize()
+	. = ..()
+	GLOB.gamemode_key_items += src
+
+
+/obj/item/disk/nuclear/crash/Destroy()
+	. = ..()
```

Comment:
Swap these two and make it a return. Destroy should always call parent last unless in very specific circumstances, or issues may arise